common/src/primitives/height.rs

```diff
@@ -27,11 +122,69 @@ impl BlockHeight {
         MAX
     }
 
-    pub fn inner(self) -> u64 {
-        self.0
+    pub fn checked_add(&self, rhs: HeightIntType) -> Option<Self> {
+        self.0.checked_add(rhs).map(Self::new)
+    }
+
+    pub fn next_height(&self) -> BlockHeight {
+        BlockHeight(self.0.checked_add(1).expect("Block height overflow"))
+    }
+}
+
+/////////////////////////////
+
+#[derive(Debug, Copy, Clone, PartialOrd, Ord, PartialEq, Eq, Encode, Decode)]
+pub struct BlockDistance(DistanceIntType);
+
+impl BlockDistance {
+    pub fn new(distance: DistanceIntType) -> Self {
+        Self(distance)
+    }
+}
+
+// Display should be defined for thiserr crate
+impl fmt::Display for BlockDistance {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        self.0.fmt(f)
+    }
+}
+
+impl From<BlockDistance> for DistanceIntType {
+    fn from(block_height: BlockDistance) -> DistanceIntType {
+        block_height.0
     }
+}
+
+impl From<DistanceIntType> for BlockDistance {
+    fn from(w: DistanceIntType) -> BlockDistance {
+        BlockDistance(w)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn basic_arithmetic() {
+        let h1 = BlockHeight::new(5);
```

Comment:
Maybe call this `h5` so it's clearer what's going on.  Or `h_5` for consistency with the likes of `d_4` below.

Reply:
h1 is some height. h_5 is height of 5. I was hoping this would be clear from the context.